### ISSUE TITLE
centos7.3  :: vnc on localhost locking up  tigervnc-server 1.8.0

### ISSUE THREAD

**Issue by cuky23** [TigerVNC/tigervnc#544]
centos7 repo version for me will lock up randomly when using it with packer/qemu.

```
Name        : tigervnc-server
Arch        : x86_64
Version     : 1.8.0
Release     : 1.el7
Size        : 213 k
Repo        : base/7/x86_64
```
This doesnt happen on the older version on ubuntu1604 tightvnc Version: 1.3.10-0ubuntu3

The service is kicked off manually by qemu, sometimes it just locks up. When the installer wants to undertake many backspaces at once.( as per ubuntu netinstall).  This only happens for me when i have a Centos7 Host using qemu/vnc connection to ubuntu16. el6 and el7 guests dont need a lot of backspaces to replace the linux boot line, so are not effected.  

it also may be worth mentioning that when i force the X connection to  real window it doesnt happen, this lockup happens when I send the vnc to localhost:59XX

in packer we call this headless mode.  headless: true, this allows the vnc terminal to be sent to localhost

**Comment by CendioOssman** [comment]
I'm not familiar with that QEMU integration. Could you start vncserver by hand and see if the issue appears there as well?

> it also may be worth mentioning that when i force the X connection to real window it doesnt happen, this lockup happens when I send the vnc to localhost:59XX

I'm not sure what you mean here?

**Comment by bphinz** [comment]
I’m also very confused by what you are describing.  When you say that it
does not happen with tightvnc on ubuntu 16.04 are you talking about the
host or the guest?

-brian

**Comment by cuky23** [comment]
centos73 host ubuntu16 guest

**Comment by bphinz** [comment]
On Tue, Nov 21, 2017 at 9:46 AM cuky <notifications@github.com> wrote:

> centos73 host ubuntu16 guest
>
But you’re saying that on a ubuntu16 host (albeit with tightvnc) it does
not happen, correct?  If that’s the case, would you be able to help us
narrow the scope of the problem by testing the same version of tigervnc
(starting with the latest integration builds) on both types of host?
RedHat applies a lot of customizations to both our source as well as the
xorg sources, so this test might help to rule some of that out.  We have
prebuilt packages for both CentOS 7 and Ubuntu 16.04 available. Thanks.

**Comment by cuky23** [comment]
I can indeed look into this was about to work out how to get TigerVNC or tightvnc on both at the same versions.

However its gonna be hard getting the old one of out ubuntu considering the dependency lists :( 
Take it this one will work on ubuntu and centos? https://bintray.com/tigervnc/stable/download_file?file_path=tigervnc-1.8.0.x86_64.tar.gz

**Comment by CendioOssman** [comment]
I'm still confused as to how you have things set up. Is it something like this:

 * Machine A running a TigerVNC client.
 * Machine B running a TigerVNC server on CentOS 7
 * Virtual machine C running Ubuntu 16.04 as a guest on Machine B using QEMU

And then you are running virt-manager inside the TigerVNC session on machine B to connect to machine C?

**Comment by cuky23** [comment]
Machine A (centos7.4) running TigerVNC using vnc via qemu
Machine B (ubuntu1604) running tightvnc  using vnc via qemu

When I use vncviewer on the host starting qemu to connect to see how the install is going. When the guest VM is ubuntu, the boot process involves undertaking a large of '<bs>' backspace chars.  (about 20 of them at least)  the VNC terminal seems to lock up and not complete all the backspaces and therefor fail.  This only seems to happen when I make the vnc server start on localhost.

installing centos69 and centos74 guests this doesnt happen, also doesnt happen if u dont re-direct qemu to VNC on localhostl and let it launch an xwindow.

When I tested TigerVNC on Machine B I saw the same lockup happen, so I am assuming that as this happened on two different OS's with the same version of TigerVNC, it may be an issue with buffering of the keyboard strokes or something.
